### PR TITLE
Only route to authenticated pages

### LINKED ISSUE #12: Navigating through other pages from the main project page adds new packages to the project
Steps to reproduce:
 - go to the main dashboard page
 - then click on a "dashboard" link under an active package
 => the list of active packages (and available packages) doubles

This also happens if I go to a package configuration page and then come back to the dashboard page.

### LINKED ISSUE #13: After the user has logged out, using the back/forth browser buttons still shows content pages instead of the login page
(If however I refresh the page, the authentication page appears correctly, so it's only with the back (and forth) browser buttons that maybe interact strangely with our routing logic?)

### PR DESCRIPTION
**Pull request by ktkization**
Fixes #12 
Fixes #13

### CODE REVIEW

File: webapp/lib/router.dart [hunk 1]
```diff
@@ -1,28 +1,46 @@
...
+  Map<String, Route> _routes;
+  Route _authRoute;
 
   Router() {
     _routes = {};
   }
 
-  void addHandler(String route, Function callback) {
-    _routes[route] = callback;
+  void addHandler(Route route, {bool isAuthRoute = false}) {
```
Comment (marianamarasoiu) [inline webapp/lib/router.dart]: Let's have a separate method for the auth page instead of passing this boolean, e.g. `addAuthHandler`

File: webapp/lib/router.dart [hunk 2]
```diff
@@ -1,28 +1,46 @@
...
+    var targetRoute = _routes[path];
+    if (targetRoute != null) {
+      if (controller.signedInUser != null && targetRoute.path != _authRoute.path) {
+        targetRoute.handler();
+        window.location.hash = targetRoute.path;
+      } else if (controller.signedInUser == null) {
+        _authRoute.handler();
+        window.location.hash = targetRoute.path;
+      }
+    }
```
Comment (marianamarasoiu) [inline webapp/lib/router.dart]: this multi-level if/if/else if block is omitting some pages, and is a bit hard to follow. I suggest something like the following (not tested + needs other changes in the code + see comments inline):

```suggestion
    if (targetRoute == null) {
      _defaultRoute.handler(); // this needs adding in the same was as we add _authRoute, should point to the dashboard page
      window.location.hash = _defaultRoute.path;
      return;
    }
    if (controller.signedInUser == null) {
      _authRoute.handler();
      window.location.hash = targetRoute.path; // shouldn't this be _authRoute.path ?
      return;
    }
    targetRoute.handler();
    window.location.hash = targetRoute.path;
```

**Review by marianamarasoiu — APPROVED** [review]
Thanks @ktkization  - a couple of comments for you

Can you also please merge master in? Merging the PR branch didn't seem to have had the desired effect. thanks!

File: webapp/lib/router.dart (hunk 1 above)
Comment (ktkization) [reply webapp/lib/router.dart]: Added `addAuthHandler`

File: webapp/lib/router.dart (hunk 2 above)
Comment (ktkization) [reply webapp/lib/router.dart]: Modified this a bit.

**Review by marianamarasoiu — APPROVED** [review]
LGTM!
